mountpoint-s3/src/prefetch.rs

```diff
@@ -206,22 +248,19 @@ pub struct PrefetchGetObject<Stream: ObjectPartStream, Client: ObjectClient> {
     client: Arc<Client>,
     part_stream: Arc<Stream>,
     config: PrefetcherConfig,
-    // Invariant: the offset of the first byte in this task's part queue is always
-    // self.next_sequential_read_offset.
-    current_task: Option<RequestTask<Client::ClientError>>,
-    // Currently we only every spawn at most one future task (see [spawn_next_request])
-    future_tasks: VecDeque<RequestTask<Client::ClientError>>,
+    backpressure_task: Option<RequestTask<Client::ClientError>>,
     // Invariant: the offset of the last byte in this window is always
     // self.next_sequential_read_offset - 1.
     backward_seek_window: SeekWindow,
     bucket: String,
     object_id: ObjectId,
     // preferred part size in the prefetcher's part queue, not the object part
     preferred_part_size: usize,
+    current_read_window_size: usize,
+    next_read_window_size: Option<usize>,
```

Comment:
May it make things simpler if we'd always reason in terms of absolute offsets, including this `next_read_window_size` variable?

I understand that now you use `next_read_window_size` as a relative value, which will be added to the current allowed offset. The problem is that since we mix absolute offsets and relative values, the code is hard to follow (e.g. you need to look around to understand what's `fn prepare_next_read_window` is doing): you always need to keep in mind what is relative and what is absolute. 

May be if we'd just record the last allowed offset that we've set for the request and the previous one, we'd still be able to calculate the value to use in `current_task.increment_read_window(..)`?

As aside -- didn't get why the calculation of `next_read_window_size` is separated from the call to `increment_read_window`. This creates the need for this `PrefetchGetObject::next_read_window_size` field, which could've been just a local variable.